frappe/database/database.py

```diff
@@ -835,8 +839,8 @@ def get_single_value(self, doctype, fieldname, cache=True):
 		if doctype not in self.value_cache:
 			self.value_cache[doctype] = {}
 
-		if cache and fieldname in self.value_cache[doctype]:
-			return self.value_cache[doctype][fieldname]
+		if cache and str(fieldname) in self.value_cache[doctype]:
```

Comment:
Why was this cast required?

Reply:
I was actually considering to defer that change. This value can be the one of a dynamic linked field on which to filter, `{"reference_document": DocRef}`. But let's defer it maybe and introduce it with a concrete example / need in the coming days/weeks.

Reply:
Yep, that would be better.

Reply:
```python
frappe.db.get_value(doctype, docname, "name", cache=True)
```

`docname` is a `DocRef` but called `filters` inside `frappe/database/database.py`